includes/class-wc-stripe-intent-controller.php

```diff
@@ -1174,6 +1175,18 @@ public function create_and_confirm_setup_intent_ajax() {
 		} catch ( WC_Stripe_Exception $e ) {
 			WC_Stripe_Logger::log( 'Failed to create and confirm setup intent. ' . $e->getMessage() );
 
+			/**
+			 * Filter the rate limit delay after a failure adding a payment method.
+			 *
+			 * @since 9.7.0
+			 *
+			 * @param int $rate_limit_delay The rate limit delay in seconds.
+			 * @param WC_Stripe_Exception $e The exception that occurred.
+			 */
+			$rate_limit_delay = apply_filters( 'wc_stripe_add_payment_method_on_error_rate_limit_delay', 10, $e );
```

Comment:
I used 10 seconds here as the default used when success (20 seconds) felt too long, but we may want to change it to something lower, like 5 seconds.